src/games/TicTacToe.ts

```diff
@@ -0,0 +1,370 @@
+        this.printGameBoard(game);
+
+        const winner = this.getGameWinner(game);
+        if (winner) {
+            this.handleEndOfGame(game, winner);
+            return;
+        }
+
+        this.doBottyMove(game);
+    }
+
+    private handleEndOfGame(game: TicTacToeGame, winner: string) {
+
+        const oplayer = game.channel.members.find(m => m.id === game.Oplayer.id);
+        const xplayer = game.channel.members.find(m => m.id === game.Xplayer.id);
+
+        const high = game.Xplayer.id.localeCompare(game.Oplayer.id) >= 0;
+        const highP = high ? { name: xplayer.user.username, id: xplayer.id } : { name: oplayer.user.username, id: oplayer.id };
+        const lowP = high ? { name: oplayer.user.username, id: oplayer.id } : { name: xplayer.user.username, id: xplayer.id };
+
+        if (winner === game.Oplayer.id) {
+            if (high) {
+                this.scores[game.scoreKey].highWins++;
+            } else {
+                this.scores[game.scoreKey].lowWins++;
+            }
+        } else if (winner === game.Xplayer.id) {
+            if (high) {
+                this.scores[game.scoreKey].lowWins++;
+            } else {
+                this.scores[game.scoreKey].highWins++;
+            }
+        } else if (winner === "-1") {
+            this.scores[game.scoreKey].draws++;
+        }
+
+        game.isActive = false;
+        game.channel.send("The game has ended!\nThis channel will be deleted in 30 seconds.");
+        this.printScores(game.channel, highP, lowP);
+        game.deleteTimeout = setTimeout(() => {
+            this.games = this.games.filter(g => g !== game);
+            game.channel.delete();
+        }, 30 * 1000);
+    }
+
+    private printScores(channel: Discord.TextChannel, highP: { name: string, id: string }, lowP: { name: string, id: string }) {
+        const key = highP.id + "_" + lowP.id;
+        const highScore = this.scores[key].highWins;
+        const lowScore = this.scores[key].lowWins;
+        const drawScore = this.scores[key].draws;
+        channel.send(`Scores are: ${highP.name} (${highScore}) - ${drawScore} - (${lowScore}) ${lowP.name}`);
+    }
+
+    /**
+     * returns "-1" for a draw
+     * returns undefined for unfinished
+     * otherwise the playerid of the winner
+     */
+    private getGameWinner(game: TicTacToeGame) {
+
+        if (this.testPlayerWin(game, TicTacToeType.X)) {
+            return game.Xplayer.id;
+        }
+
+        if (this.testPlayerWin(game, TicTacToeType.O)) {
+            return game.Oplayer.id;
+        }
+
+        for (let i = 1; i < 10; i++) {
+            if (game.board[i] === undefined) {
+                return undefined;
+            }
+        }
+
+        return "-1";
+    }
+
+    private testPlayerWin(game: TicTacToeGame, player: TicTacToeType) {
+        const isAllPlayer = (...args: number[]) => args.filter(x => game.board[x] === player).length === args.length;
```

Comment:
Can probably just use `.every` here instead of `filter(x) === length`, but thats my bad.